### PR TITLE
Quick switch folders with keyboard shortcut

### PR DESCRIPTION
**Pull request by jainil**
This adds a keyboard shortcut to quickly switch between folders.

### CODE REVIEW

File: Default.sublime-keymap [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[
+    { "keys": ["super+shift+o"], "command": "side_bar_folders_quick_switch" }
```
Comment (facelessuser) [inline Default.sublime-keymap]: Please don't add shortcuts by default as they may clash with other shortcuts.

Comment (Jainil) [inline Default.sublime-keymap]: I have added instructions to the readme for changing the shortcut. Are you suggesting that I should  remove the keymap file all together and leave it up to the user to configure the shortcut after installing the plugin?

Comment (facelessuser) [inline Default.sublime-keymap]: > Are you suggesting that I should remove the keymap file all together and leave it up to the user to configure the shortcut after installing the plugin?

Basically, yeah.  Just document an example.

For instance, I have a personal shortcut already configured for <kbd>super</kbd> + <kbd>shift</kbd> + <kbd>o</kbd>.  There are frankly too many plugins that assume everyone wants their shortcuts to be defined by the plugins.  There is no way for a plugin author to be aware of all the plugins trying to use the same shortcut.  It causes confusion.  It is much better just to provide an example keymap and let the user do it.

If this got pulled as it is now, it is guaranteed to break me and probably a number of others.  I think it is much better to break no one, and the people who want a shortcut can simply copy and paste (and possibly tweak) an example if desired.

The only exception for including keymaps is if they have a context assigned to them that limits them to a very particular setting.  For instance, I have plugins that I have written that enable keymaps only within a special view that is plugin specific, or a keymap that is only enabled in the quick panel for a specific command.  By narrowing the scope, I conflict with nothing, but global keymaps will certainly conflict with a number of peoples setups.

Comment (facelessuser) [inline Default.sublime-keymap]: As a compromise, you could expose the command in the command palette to give users something useable out of the box and provide an example to get immediate access via a keybinding.

Comment (Jainil) [inline Default.sublime-keymap]: Thanks for the suggestion, I'll update the PR with the changes soon.

File: SideBarFolders.py [hunk 2]
```diff
@@ -342,6 +342,51 @@ def run(self):
...
+	def run(self):
+		folder_list = list(map(lambda folder: folder['path'], s.get('folders', [])))
+		folder_list.reverse()
+		self.display_list(folder_list)
+
+	def display_list(self, list):
+		self.window.show_quick_panel(list, self.on_done)
+
+	def on_done(self, index = -1, append = False):
+		print(index)
```
Comment (facelessuser) [inline SideBarFolders.py]: You will probably want to make sure you have removed all debug statements.

File: SideBarFolders.py [hunk 3]
```diff
@@ -342,6 +342,51 @@ def run(self):
...
+		folder = (Pref.folders[::-1])[index]
+		if self.audit_folder(folder, index):
+			return
+		project = get_project_data(Window())
```
Comment (facelessuser) [inline SideBarFolders.py]: Did you mean self.window?  Maybe I am missing something here.

Comment (titoBouzout) [inline SideBarFolders.py]: In some situations that I cannot remember window and view are incorrect, so I always use sublime.active_window() and sublime.active_view() which I can certainly be sure that the value is correct.

**Comment by titoBouzout** [comment]
I merged this manually, many thanks @Jainil !

**Pull request closed without merging**